numba/np/random/generator_methods.py

```diff
@@ -819,3 +820,95 @@ def impl(inst, n, p , size=None):
                 out[i] = random_negative_binomial(inst.bit_generator, n, p)
             return out
         return impl
+
+
+@overload_method(types.NumPyRandomGeneratorType, 'noncentral_chisquare')
+def NumPyRandomGeneratorType_noncentral_chisquare(inst, df, nonc, size=None):
+    check_types(df, [types.Float, types.Integer, int, float], 'df')
+    check_types(nonc, [types.Float, types.Integer, int, float], 'nonc')
+    if isinstance(size, types.Omitted):
+        size = size.value
+
+    @register_jitable
+    def check_arg_bounds(df, nonc):
+        if df <= 0:
+            raise ValueError("df <= 0")
+        if nonc < 0:
+            raise ValueError("nonc < 0")
+
+    if is_nonelike(size):
+        def impl(inst, df, nonc, size=None):
+            check_arg_bounds(df, nonc)
+            return np.float64(random_noncentral_chisquare(inst.bit_generator,
+                                                          df, nonc))
+        return impl
+    else:
+        check_size(size)
+
+        def impl(inst, df, nonc, size=None):
+            out = np.empty(size, dtype=np.float64)
+            for i in np.ndindex(size):
+                out[i] = random_noncentral_chisquare(inst.bit_generator,
+                                                     df, nonc)
+            return out
+        return impl
+
+
+@overload_method(types.NumPyRandomGeneratorType, 'noncentral_f')
+def NumPyRandomGeneratorType_noncentral_f(inst, dfnum, dfden, nonc, size=None):
+    check_types(dfnum, [types.Float, types.Integer, int, float], 'dfnum')
+    check_types(dfden, [types.Float, types.Integer, int, float], 'dfden')
+    check_types(nonc, [types.Float, types.Integer, int, float], 'nonc')
+    if isinstance(size, types.Omitted):
+        size = size.value
+
+    @register_jitable
+    def check_arg_bounds(dfnum, dfden, nonc):
+        if dfnum <= 0:
+            raise ValueError("df <= 0")
+        if dfden <= 0:
+            raise ValueError("df <= 0")
```

Comment:
```suggestion
        if dfnum <= 0:
            raise ValueError("dfnum <= 0")
        if dfden <= 0:
            raise ValueError("dfden <= 0")
```
that testing didn't pick this up suggests that the tests are not sufficiently specific or this is not being tested.